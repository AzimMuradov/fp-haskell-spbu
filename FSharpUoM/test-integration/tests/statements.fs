let x = 4;
let y = x * 2;
let m = x + y;
m;; // Right: "12"

let f x = x + 1;; // Right: "int -> int"

let f x = begin
    2 + 3;
    let p = x * 2 in 
        p + 2;
    let g m = m * 10 in
        g x
end;
f 10;; // Right: "100"

let rec fib y n = if n <= 1 
    then 1 
    else fib (n - 1) y + fib (n - 2) y; 

fib 1;; // Right: "int -> int"

[<Measure>] type m;; 
// Right: 
// "[<Measure>]
// type m = m"

[<Measure>] type m;
[<Measure>] type sec;
[<Measure>] type speed = m / sec;
3<m> / 2<sec> + 2<speed>;; // Right: "3<m/(sec)>"



