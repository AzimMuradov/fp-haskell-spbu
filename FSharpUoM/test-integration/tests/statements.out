val it: int = 12
val f: int -> int
val it: int = 100
val it: int -> int
[<Measure>]
type m = m
val it: int<m/(sec)> = 3
